### PR TITLE
[CONTENT] GSoC 2020 RAPPS Stage 1 & 2

### PR DESCRIPTION
**Pull request by kernelbin**

### CODE REVIEW

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+author:      "kernel.bin"
+date:        2020-08-05
+tags:        [ "gsoc", "gsoc-2020", "rapps" ]
+banner:      "/img/blogs/gsoc2020-rapps-stage1&2.png"
+---
+
+Hi, I'm He Yang (@kernel.bin), one of the GSoC-2020 students of ReactOS.
+Two months have passed since GSoC started, and during these two months, I've made some progress on ReactOS Application Manager (RAPPS), the project I'm working on.
+
+In this blog, I will list out what I've accomplished in these two months, and what I'm planning to do recently.
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:12]:
```suggestion
In this blog, I will list out what I've accomplished in these two months, and what I'm planning to do in the next month.
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## The work I've done in Stage 1&2
+
+### Stage 1
+
+During stage 1, I mainly focus on screenshot and icon support and some improvements.
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:18]:
```suggestion
During stage 1, I mainly focused on screenshot and icon support and various small improvements.
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+#### work done:
+- Add support for screenshots. One can specify URLs to screenshot images in rapps-db .txt file. It will be displayed at the left side of the rich-edit, automatically sized. when downloading / download failed, a placeholder will be displayed. (Currently, only the first image will be displayed).
+
+- Improve icon support. now icon can be specified in rapps-db .txt file too. (the original design have problems if some special char exists in software's name)
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:23]:
```suggestion
- Improve icon support. Now icon can be specified in rapps-db .txt file too. (the original design had problems if some special char exists in the software name)
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+- Some other invisible improvements (Introducing AsyncInet, path-handling improvement, and so on) and some refactoring.
+
+### Stage 2
+
+During stage 2, I mainly focus on improving listview and code about it. I made a big refactor on it, and improve UI by a little.
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:29]:
```suggestion
During stage 2, I mainly focus on improving listview and related code. This turned out the be a pretty big refactor, and it improved the UI a bit.
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+#### work done:
+- Move ToolBar and SearchBar to AppView, and remove "settings" and "exit" button.
+
+- Add a ComboBox in ToolBar, enabling users to choose how to view apps. (Supports details mode and list mode. Tile mode is only supported on Windows)
+
+- Some small bugfix and improvements. (Correctly handle Select All / Deselect All and Column-Click. Handle double-click and RightClick->Install in listview more reasonably and so on)
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:36]: You might want to explain that Column-Click now allows sorting, this is a nice usability improvement!

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+- Some other invisible improvements and some refactoring. (Make code around listview, class CAvailableApps, CInstalledApps more clear，handle WOW64 registry correctly)
+
+**screenshots**
+![RAPPS screenshot](/img/blogs/gsoc2020-rapps-stage1&2.png)
+
+## Short-term Plan
+
+listed below are the work I'm planning to do recently.
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:45]:
```suggestion
Listed below is the work I'm planning to do next.
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+- refactor download dialog box and code about downloading. Support concurrent download, reuse the code (```AsyncInet.cpp```) if possible.
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:47]:
```suggestion
- refactor download dialog box and code about downloading. Support concurrent download, reuse the code from (```AsyncInet.cpp```) if possible.
```

File: content/blogs/GSoC-2020-rapps-stage1&2.md (hunk 5 above)
Comment (kernelbin) [reply content/blogs/GSoC-2020-rapps-stage1&2.md:36]: > You might want to explain that Column-Click now allows sorting, this is a nice usability improvement!

No, Column-Click sorting was supported before 😅 (with some bugs when changing to sort with another column or switching category).
I just got these bugs fixed🤣.

BTW, the listview still has something that can be improved.
when sorting, I would prefer non-English language to be sorted by pronunciation. but sort by WCHAR (I mean, UTF16BE) will not lead to this. It just looks like randomly sorted. I will probably fix this later. (I guess sort by UTF8 can fix this?)

Comment (learn-more) [reply content/blogs/GSoC-2020-rapps-stage1&2.md:36]: TBH I don't know enough about sorting non-ascii text.

File: content/blogs/GSoC-2020-rapps-stage1&2.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
+---
+title:       "Rapps Enhancement stage 1 & 2 - GSoC 2020"
+author:      "kernel.bin"
+date:        2020-08-05
```
Comment (learn-more) [inline content/blogs/GSoC-2020-rapps-stage1&2.md:4]:
```suggestion
date:        2020-08-08
```

**Pull request merged**